src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -77,32 +80,124 @@ public DockerBuilder(String repoName, String repoTag, boolean skipPush, boolean
+        this.server = server;
+        this.registry = registry;
+        this.repoName = repoName;
+    }
+
+    public DockerServerEndpoint getServer() {
+        return server;
+    }
+
+    public DockerRegistryEndpoint getRegistry() {
+        // coming from an older version <1.0 ? let's try to parse the registry
+        // TODO should do this on the descriptor readResolve ?
+        if (registry == null) {
+            registry = DockerRegistryEndpoint.fromImageName(repoName, null);
+            if (registry.getUrl() != null) {
+                repoName = repoName.substring(repoName.indexOf('/') + 1); // take out the host:port part
+                logger.log(
+                        Level.WARNING,
+                        "Using registry from old configuration field, you may need to configure credentials in the build step: {0} {1}",
+                        new String[] { registry.getUrl(), repoName });
+            }
+        }
+
+        return registry;
+    }
+
+    public String getRepoName() {
+        return repoName;
+    }
+
+    public boolean isNoCache() {
+        return noCache;
+    }
+
+    @DataBoundSetter
+    public void setNoCache(boolean noCache) {
+        this.noCache = noCache;
+    }
+
+    public boolean isForcePull() {
+        return forcePull;
+    }
+
+    @DataBoundSetter
+    public void setForcePull(boolean forcePull) {
+        this.forcePull = forcePull;
+    }
+
+    public String getDockerfilePath() {
+        return dockerfilePath;
+    }
 
+    @DataBoundSetter
+    public void setDockerfilePath(String dockerfilePath) {
+        this.dockerfilePath = dockerfilePath;
+    }
+
+    public boolean isSkipBuild() {
+        return skipBuild;
+    }
+
+    @DataBoundSetter
+    public void setSkipBuild(boolean skipBuild) {
+        this.skipBuild = skipBuild;
+    }
+
+    public boolean isSkipDecorate() {
+        return skipDecorate;
+    }
+
+    @DataBoundSetter
+    public void setSkipDecorate(boolean skipDecorate) {
+        this.skipDecorate = skipDecorate;
+    }
+
+    public String getRepoTag() {
+        return repoTag;
+    }
+
+    @DataBoundSetter
+    public void setRepoTag(String repoTag) {
+        this.repoTag = repoTag;
+    }
+
+    public boolean isSkipPush() {
+        return skipPush;
+    }
+
+    @DataBoundSetter
+    public void setSkipPush(boolean skipPush) {
+        this.skipPush = skipPush;
+    }
+
+    public boolean isSkipTagLatest() {
+        return skipTagLatest;
+    }
+
+    @DataBoundSetter
+    public void setSkipTagLatest(boolean skipTagLatest) {
+        this.skipTagLatest = skipTagLatest;
+    }
+
+    /**
+     * Fully qualified repository/image name with the registry url in front
+     * @return ie. docker.acme.com/jdoe/busybox
+     * @throws IOException
+     */
+    public String getRepo() throws IOException {
+        return getRegistry().imageName(repoName);
+    }
 
 
     private boolean defined(String s) {
     	return s != null && !s.trim().isEmpty();
```

Comment:
not touched in this PR and seems a bit different that it is checking, not really need the result of the trim